cmd/runtimetest/main.go

```diff
@@ -370,6 +371,52 @@ func validateMountsExist(spec *rspec.Spec) error {
 	return nil
 }
 
+func getActualCgroupPath() (string, error) {
+	f, err := os.Open("/proc/self/cgroup")
+	if err != nil {
+		return "", err
+	}
+	defer f.Close()
+
+	s := bufio.NewScanner(f)
+	for s.Scan() {
+		if err := s.Err(); err != nil {
+			return "", err
+		}
+
+		text := s.Text()
+		parts := strings.Split(text, ":")
+		return parts[2], nil
```

Comment:
With 946b75e → 8d6fcb6 it looks like you are no longer blessing `devices`, but are instead returning the first entry you find.  That's fine as long as it happens to be one of the controllers managed by the runtime, but opencontainers/runtime-spec#493 only requires the runtime to manage controllers needed to enforce `linux.resources`.  If the first entry happens to *not* be one of those controllers, the path you're extracting here may not be the unified path used by controllers that the runtime is managing.  I think the logic should be:

> For all controllers needed for `linux.resources` (may be an empty set), ensure that they have the same path.  Return that path for `validateCgroupsPath`.